### PR TITLE
Feature admin register

### LINKED ISSUE #3: Admin registration
The super-admin can create regular admins.

### PR DESCRIPTION
**Pull request by Vasilijez**
Closes #3.

After logging in, a super admin can add a new regular admin using a straightforward registration form.

Since adding a new admin relies on the process of registering a new user, unit and integration tests have been implemented for both functionalities.

### CODE REVIEW

File: client/src/pages/UserAdminRegistration.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                                <Form.Control
+                                    type="password"
+                                    name="confirmPassword"
+                                    placeholder="Confirm admin password"
+                                    value={formData.confirmPassword}
+                                    onChange={handleChange}
+                                    required
+                                />
+                            </Col>
+                        </Form.Group>
```
Comment (magley) [inline client/src/pages/UserAdminRegistration.tsx:117]: I think password confirmation isn't necessary when in this case, because we can assume that the super-admin is extra careful when creating admin accounts.
There's a whole debate over whether password confirmation is a good idea in the first place, both security-wise and also from a UI point of view.

If you want to keep it, that's fine. I guess it doesn't matter much.

File: server/app/tests/test_user.py [hunk 2]
```diff
@@ -58,4 +66,151 @@ def test_change_password_new_password_is_same_as_current_password(user_service,
...
+        mock_user_repo.set_role.return_value = user_end
+
+        def set_role(user: User, role: UserRole):
+            user.sqlmodel_update({"role": role})
+            return user
+        mock_user_repo.set_role.side_effect = set_role
+
+        result = user_service.add_admin(dto)
+        
+        assert result.role == UserRole.admin
```
Comment (magley) [inline server/app/tests/test_user.py:118]: You don't have to (shouldn't?) mock `add()` here.

When unit testing, you mock _external dependencies_. `user_repo` is an external dependency which is why we mock it. When testing a class, you don't mock other parts of the class because those are _implementation details_.

Should you decide to refactor your code tomorrow (e.g. have `add_admin` not rely on `add` anymore), the test won't pass anymore. But the whole point of refactoring is to change the structure of the code without changing its behavior, so the test _must_ pass.

The other reason is that we _want_ `add_admin` to fail when `add` fails. We've discussed whether you should test `add_admin` as thoroughly as `add`, but there's really no need for such trivial code. This unit test cares about input (parameters) and output (return value).

There's a good [thread](https://stackoverflow.com/questions/57235607/how-to-mock-a-method-that-calls-other-methods-that-are-present-in-the-same-class) on StackOverflow discussing this.

File: server/app/tests/test_user.py [hunk 3]
```diff
@@ -58,4 +66,151 @@ def test_change_password_new_password_is_same_as_current_password(user_service,
...
+            "password": "Password1"
+        }
+        response = client.post("/api/v1/users/register-admin", json=data, headers=header)
+        assert response.status_code == 200
+
+        created_admin = response.json()
+        assert created_admin["username"] == data["username"]
+        assert created_admin["role"] == "admin"
+
+def test_add___integration():
```
Comment (magley) [inline server/app/tests/test_user.py:187]: Add two more test cases here: duplicate username and duplicate email (checking if the API will return a status code of 400).

**Review by magley — CHANGES_REQUESTED** [review]

File: client/src/pages/UserAdminRegistration.tsx (hunk 1 above)
Comment (Vasilijez) [reply client/src/pages/UserAdminRegistration.tsx:117]: I considered it from a UI perspective. While the super-admin is an advanced user and may not necessarily need this feature, he is still human and prone to making typos, just like any regular user. For now, I’ve decided to keep it.

File: server/app/tests/test_user.py (hunk 3 above)
Comment (Vasilijez) [reply server/app/tests/test_user.py]: I initially chose not to include these tests, as the unit tests already cover this functionality, and I wanted to focus on testing the core functionality through integration tests. However, I will add these tests as suggested.
